src/main/java/com/iexec/core/task/update/TaskUpdateManager.java

```diff
@@ -54,21 +56,24 @@ class TaskUpdateManager  {
     private final WorkerService workerService;
     private final BlockchainAdapterService blockchainAdapterService;
     private final SmsService smsService;
+    private final SmsClientProvider smsClientProvider;
 
     public TaskUpdateManager(TaskService taskService,
                              IexecHubService iexecHubService,
                              ReplicatesService replicatesService,
                              ApplicationEventPublisher applicationEventPublisher,
                              WorkerService workerService,
                              BlockchainAdapterService blockchainAdapterService,
-                             SmsService smsService) {
+                             SmsService smsService,
+                             SmsClientProvider smsClientProvider) {
```

Comment:
I find strange at that level to to inject twice something related to SMSs. Maybe you could relocate `SmsClientProvider`  part in `SmsService` then expose a new method from there `SmsService#isClientReady(String chainTaskId)` (?)

Reply:
Good idea, please see https://github.com/iExecBlockchainComputing/iexec-core/pull/519/commits/d92a8733145de56e753d930854748ac0524aa300.